src/app/model/ChessEngine.ts

```diff
@@ -67,4 +72,49 @@ export class ChessEngine implements ChessModel {
   private changeTurn(side: Side): Side {
     return side === Side.WHITE ? Side.BLACK : Side.WHITE;
   }
+
+  private willBeKingChecked(player: Player, chessboard: Chessboard, squareFrom: Square, squareTo: Square): boolean {
+    let isCheckedSquareFlag = false;
+
+    const proposedChessboard = _.cloneDeep(chessboard);
+    const { squaresWithPiece: proposedSquaresWithPieces } = proposedChessboard;
+
+    const movedPiece = chessboard.onPositionPiece(squareFrom);
+    const positionFrom = `${squareFrom.column}${squareFrom.row}`;
+    const positionTo = `${squareTo.column}${squareTo.row}`;
+    delete proposedSquaresWithPieces[positionFrom];
+    proposedSquaresWithPieces[positionTo] = movedPiece as Piece;
+
+    const allyKingPositionKey = Object.keys(proposedSquaresWithPieces).find((key) => {
+      const isKingName = proposedSquaresWithPieces[key].name === 'King';
+      const isPlayerSide = proposedSquaresWithPieces[key].side === player.side;
+      return isKingName && isPlayerSide;
+    });
+
+    if (!allyKingPositionKey) return (isCheckedSquareFlag = false);
+
+    const kingPosition = {
+      column: allyKingPositionKey[0],
+      row: Number(allyKingPositionKey[1]) as Row,
+    };
+
+    Object.keys(proposedSquaresWithPieces).forEach((key) => {
```

Comment:
Oglądaliście serię o functional programming z ostatniego działu z Node.js :) ? I ta o której przypominam kilka razy? Zróbcie to, bo za dużo robicie "na piechotę". Trzeba korzystacz mozliwosci języka.
Np. if(isEnemySide) załatwia filter itd. SPróbuj to zrobić jako pipe :)

Reply:
https://www.youtube.com/watch?v=E_BRt_fqaeA&list=PLjHmWifVUNMLR1Oz3WYK7XRLEY5LvSOPu&index=6&t=1s

Ha, oglądałem to, ale chyba dalej nie 

> Zrobie review samego kodu, ale zadanie działa trochę nie zgodnie z opisem. Choć prawda brakowało też kawałka który zrobiłeś. Co prawda moglibysmy to po refaktorze zmergowac i potem troche zmienic. Z opisu wynika, ze z metody:
> `possibleMoves(position: Square): Square[];` - tutaj nie bylo jeszcze nazwy, ale taka jest dodana w #28 zwrócić ruchy obciete o te ktore robia szach krola. No bo, tez nie mozemy wyswietlic tych, ktore powoduja szach :) Nie musimy chyba w komunikacje bledu rozrozniac z jakiego powodu ten ruch jest niemozliwy.
> I wtedy w move mozeby po prostu uzywac possibleMoves z chessModel zamiast z piece sprawdzajac `canMoveOnSquare`. Tak samo jak dodamy roszade czy bicie w przelocie. Bedzie to ogarniac chessEngine, a metoda movePiece sie nie bedzie zmieniac.
> 
> Czy to ma sens :) ?


Widzę też, że pomieszałem opis zadania. Spytałem się chłopaków, jak to zrobić najlepiej (a nie pokazałem im opisu), no i zapomniałem o tym opisie.
No mój błąd. 
Czy w takim razie mam przepisać to zgodnie z tym co w opisie taska? :)

Reply:
@PiotrWR to przyda sie cala lista :) To jest jako materialy podstawowe do node.js